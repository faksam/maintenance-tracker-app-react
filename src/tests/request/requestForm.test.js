import React from 'react';
import { shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import ConnectedRequestForm, { RequestForm } from '../../app/components/request/RequestForm';

const props = {
  fullname: 'fakunlesamuel@gmail.com',
  email: 'fakunlesamuel@gmail.com',
  password: 'Lifeisarace',
  confirmPassword: 'Lifeisarace',
  handleSignIn: jest.fn(),
  handleSignUp: jest.fn(),
  handleChange: jest.fn(),
  hideModalRequest: jest.fn(),
  createRequestAction: jest.fn(),
  request: {},
  history: {},
  user: {
    username: 'joeeasy',
    email: 'joeeasy@gmail.com',
    role: 'Admin',
    id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
    iat: 1540985284931
  },
};
const initialState = {
  auth: {
    isAuthenticated: true,
    error: {},
    user: {
      username: 'joeeasy',
      email: 'joeeasy@gmail.com',
      role: 'Admin',
      id: '6bf54873-6033-4ab0-855d-336a34e1dfbc',
      iat: 1540985284931
    }
  },
  requestReducer: {
    requests: []
  },
  errors: {},
};
const mockStore = configureMockStore([thunk]);
const store = mockStore(initialState);

describe('SignInTabForm Component', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<ConnectedRequestForm store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
  it('should set state to props', () => {
    const wrapper = shallow(<ConnectedRequestForm store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('should a create new request form', () => {
    const wrapper = shallow(<RequestForm store={store} {...props} />);
    const instance = wrapper.instance();
    const event = {
      preventDefault: jest.fn()
    };
    wrapper.setState({
      description: 'Ibadan is a wonderful Place',
      title: 'It is a neccesaarate',
    });
    instance.onSubmit(event);
  });
  it('should a simulate typing', () => {
    const wrapper = shallow(<RequestForm store={store} {...props} />);
    const instance = wrapper.instance();
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'title',
        value: 'come',
        textContent: 'nothing',
        valueAsDate: {
          toDateString: () => 'October 12, 2010',
        },
        files: ['.png', 'jpeg'],
        matches: () => true,
      },
    };
    instance.handleCloseError();
    wrapper.setState({
      description: 'Ibadan is a wonderful Place',
      title: 'It is a neccesaarate',
      errors: {
        description: 'Error',
        title: 'Error',
      }
    });
    instance.onChange(event);
  });
  it('should render without throwing an error', () => {
    const wrapper = shallow(<RequestForm store={store} {...props} />);
    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
