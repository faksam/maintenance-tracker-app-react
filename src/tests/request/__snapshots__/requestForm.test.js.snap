// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInTabForm Component should render without throwing an error 1`] = `
<RequestForm
  confirmPassword="Lifeisarace"
  createRequestAction={[Function]}
  email="fakunlesamuel@gmail.com"
  fullname="fakunlesamuel@gmail.com"
  handleChange={[MockFunction]}
  handleSignIn={[MockFunction]}
  handleSignUp={[MockFunction]}
  hideModalRequest={[MockFunction]}
  history={Object {}}
  password="Lifeisarace"
  request={Object {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  user={
    Object {
      "email": "joeeasy@gmail.com",
      "iat": 1540985284931,
      "id": "6bf54873-6033-4ab0-855d-336a34e1dfbc",
      "role": "Admin",
      "username": "joeeasy",
    }
  }
/>
`;

exports[`SignInTabForm Component should render without throwing an error 2`] = `
<div
  className="modal"
  id="createRequestModal"
>
  <div
    className="modal-content"
  >
    <div
      className="modal-header"
    >
      <span
        className="close"
        id="createRequestSpan"
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          }
        }
      >
        Ã—
      </span>
    </div>
    <div
      className="modal-body"
    >
      <div
        className="row"
      >
        <div
          className="container center"
        >
          <div
            className="row"
          >
            <h2
              className="float-left"
            >
              Create A New Request
            </h2>
          </div>
          <form
            className=" center"
            id="createRequestForm"
            method="POST"
          >
            <div
              className="row"
            >
              <div
                className="row"
              >
                <div
                  className="input col col-1"
                >
                  <input
                    id="requestTitle"
                    name="title"
                    onChange={[Function]}
                    placeholder="Request Title"
                    required={true}
                    type="text"
                    value=""
                  />
                  <div
                    className="display-none"
                    id="errorDiv"
                  >
                    <p
                      className="hide-div"
                      id="signupErrorMessage"
                    />
                  </div>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input col col-1"
                >
                  <textarea
                    className="desc-txt-area"
                    cols="50"
                    id="requestDescription"
                    name="description"
                    onChange={[Function]}
                    placeholder="Request Description"
                    required={true}
                    rows="5"
                    value=""
                  />
                  <div
                    className="display-none"
                    id="errorDiv"
                  >
                    <p
                      className="hide-div"
                      id="signupErrorMessage"
                    />
                  </div>
                </div>
              </div>
            </div>
            <br />
            <button
              className="btn-create-request"
              id="createUserRequestButton"
              onClick={[Function]}
              type="button"
            >
              Create
            </button>
          </form>
          <br />
        </div>
      </div>
    </div>
    <br />
    <br />
  </div>
</div>
`;
